### PR TITLE
Adding SNO+ and proton scattering to sntools

### PR DESCRIPTION
**Pull request by svalder**
I have added SNO+ as a detector into sntools in the form of two different geometries: inner AV filled with liquid scintillator, and external water inside the PMT support structure. This has added two new geometries to sntools called 'sphere' and hollowSphere'. The latter is two concentric spheres with the inner being empty and the outer being filled.

SNO+ also needed the addition of Liquid Alkyl-benzene (LAB) as a detector material. This has been added with the averaged chemical composition calculated through the carbon chain lengths.

Proton elastic scattering has also been implemented as an interaction channel. This is the second largest interaction channel for SNO+ and is available to liquid scintillators. This has been based off John F Beacom et al., 2002: 10.1103/PhysRevD.66.033001.

The results have been cross-checked with SNO+ preliminary event rates, as well as other an analytical event rate calculator, EstrellaNueva. 

I've not yet added to the documentation, but I'm happy to do so in the same style as already written.

Please let me know if you have any comments/questions. Cheers, Sammy

### CODE REVIEW

File: sntools/genevts.py [hunk 1]
```diff
@@ -89,11 +89,12 @@ def parse_command_line_options():
...
 
-    choices = ("ibd", "es", "o16e", "o16eb", "c12e", "c12eb", "c12nc")
+    choices = ("ibd", "es", "o16e", "o16eb", "c12e", "c12eb", "c12nc", "ep")
     parser.add_argument("-c", "--channel", metavar="INTCHANNEL", choices=choices, default="all",
                         help="Interaction channels to consider. Currently supported: inverse beta decay (ibd), \
-                              electron scattering (es), nu_e + oxygen CC (o16e), nu_e-bar + oxygen CC (o16eb), \
-                              nu_e + carbon CC (c12e), nu_e-bar + carbon CC (c12eb) and nu + carbon NC (c12nc). \
+                              electron scattering (es), proton scattering (ep), nu_e + oxygen CC (o16e), \
+                              nu_e-bar + oxygen CC (o16eb), nu_e + carbon CC (c12e), nu_e-bar + carbon CC (c12eb), \
+                              and nu + carbon NC (c12nc). \
```
Comment (JostMigenda) [inline sntools/genevts.py:97]: Minor nitpick: The order of channels `choices` should be the same as the order in the help text.

**Comment by JostMigenda** [comment]
Thanks @svalder, this is looking good!

Don’t worry about the failing tests for now—the unit tests still pass as expected, and the integration tests only fail because the new output files are being compared to a hard-coded output file which doesn’t include proton scattering events.
(Actually, we should check with @yschnellbach: I don’t think you’ve studied proton scattering for WATCHMAN-LS thus far, right? Are you happy with it [being added as part of this PR](https://github.com/JostMigenda/sntools/pull/41/commits/9fc1656560d89fdc91883436d6a7154ea49639af) or should we remove this channel from the `ls` material for the time being?)

A couple things before we can merge this:
- [ ] Please go ahead and update the documentation, yes!
- [ ] Why `ep` as the short name for proton scattering? (Is this some SNO+ convention?) To me, `ps` for proton scattering would be more intuitive (and consistent with `es` for electron scattering, too).
- [ ] What cross checks have you done on the proton scattering implementation, beyond comparing event counts with other software? (That by itself is a good sign, but cross section code is notoriously tricky and can have more subtle bugs. I can take a look at the code & paper side by side in the next few days, but additional checks would be good.)
- [ ] Can you add a few unit tests for the proton scattering channel? (See `/tests/test_*.py` for examples for other channels.)
- [ ] Regarding tests, let’s wait for Yan-Jie’s response.

**Comment by yschnellbach** [comment]
Hi, we haven't studied proton scattering for WATCHMAN-LS so far, but looking over the PR, I see no reason to omit it from the `ls` material. Looks good and useful to me.

**Comment by JostMigenda** [comment]
Okay, I’ve just pushed a commit to update the integration test. With that, this PR is complete. 🎉 Thank you so much @svalder for all your work on this!

**Pull request merged**